lib/ssz.ex

```diff
@@ -139,4 +164,349 @@ defmodule Ssz do
 
   @spec decode_u256(binary) :: non_neg_integer
   def decode_u256(num), do: :binary.decode_unsigned(num, :little)
+
+  ##### elixir native ssz
+  @bytes_per_length_offset 4
+  @bytes_per_chunk 32
+  @bits_per_byte 8
+
+  ### ENCODE ######
+  def serialize(%struct{} = map) do
+    schemas = struct.schema()
+
+    serialized =
+      schemas
+      |> Enum.map(fn schema ->
+        key = Enum.at(Map.keys(schema), 0)
+        metadata = Map.get(schema, key)
+        value = Map.get(map, key)
+        {:ok, serialized} = serialize(value, metadata)
+        serialized
+      end)
+      |> :binary.list_to_bin()
+
+    {:ok, serialized}
+  end
```

Comment:
I think this should be the inverse (general to specific):

```elixir
def serialize(%struct{} = map), do: serialize(map, struct.schema())

def serialize(value, %{type: :struct}) do
  serialized =
    schemas
    |> Enum.map(fn schema ->
      key = Enum.at(Map.keys(schema), 0)
      metadata = Map.get(schema, key)
      value = Map.get(map, key)
      {:ok, serialized} = serialize(value, metadata)
      serialized
    end)
    |> :binary.list_to_bin()
end
```

Also, I think we should change `schema()` to return `%{type: :struct, schema: <current schema()>}`, since we may want to have non-SSZ-container structs like `SszTypes.Transaction`. wdyt? @Arkenan